root/static/scripts/edit/URLCleanup.js

```diff
@@ -312,6 +312,14 @@ function disallow(url, id) {
  *             for an auto-selected relationship type.
  */
 const CLEANUPS = {
+  '7digital': {
+    match: [new RegExp('^(https?://)?([^/]+\\.)?(7digital\\.com|zdigital\\.com\\.au)', 'i')],
+    type: LINK_TYPES.downloadpurchase,
+    clean: function (url) {
+      url = url.replace(/^https?:\/\/([^/]+\.)?7digital\.com\/yourmusic\/(.*)\/[^/]+\/?$/, 'https://$17digital.com/$2');
```

Comment:
That would be nice to clean legitimate URLs up too. An additional test would be welcome then.

Reply:
```suggestion
      url = url.replace(/^https?:\/\/([^/]+\.)?7digital\.com\/(?:yourmusic\/)?(.*)\/[^/]+\/?$/, 'https://$17digital.com/$2');
```

Reply:
This would break because of only yourmusic urls having the bit at the end that needs removing, but it might make sense to just make it two parts. Can do.

Reply:
Thanks, that should prevent any unmindful coder to make the same mistake as I just did!